warehouse/packaging/models.py

```diff
@@ -65,13 +62,18 @@ class Role(db.Model):
         CIText,
         ForeignKey("accounts_user.username", onupdate="CASCADE", ondelete="CASCADE"),
     )
-    package_name = Column(
-        Text, ForeignKey("packages.name", onupdate="CASCADE", ondelete="CASCADE")
+    project_id = Column(
+        ForeignKey("packages.id", onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
     )
 
     user = orm.relationship(User, lazy=False)
     project = orm.relationship("Project", lazy=False)
 
+    @property
+    def package_name(self):
```

Comment:
This is a problem, because we are using ``Role.project_name`` in queries as part of an ``ORDER BY``, see this traceback for ``TestUserDetail.test_gets_user``.

The correct solution depends on why we're ordering by ``Role.project_name``.. if the goal is just to provide a consistent ordering, we can change it to order by ``Role.project_id`` instead. If the goal is to explicitly order it by the project name, then you'll probably want to turn this into a [Hybrid Property](https://docs.sqlalchemy.org/en/latest/orm/extensions/hybrid.html) or a [Column Property](https://docs.sqlalchemy.org/en/latest/orm/mapped_sql_expr.html#mapper-column-property-sql-expressions). Another option would be to adjust the query to do a join manually, and then order by ``Project.name`` instead of ``Role.package_name``.